### DISCUSSION TITLE
EVGA RTX-2080 FTW3 Ultra Gaming on Asus ROG MaximusVIII formula

### DISCUSSION THREAD

**Discussion by Druboo666** [terminatorul/NvStrapsReBar#16] [Q&A]
Hi,

i have attached the report file form CPU-Z and needed data numbers also added the edited  [LocalPciGPU h](https://github.com/terminatorul/NvStrapsReBar/blob/master/ReBarDxe/include/LocalPciGPU.h).
i need to know if they are correct and as i don't have a flasher and i cant recover my motherboard from a brick if something goes wrong.
**what are ways to revert it back to stock?**
[FARADIS-ROG.txt](https://github.com/terminatorul/NvStrapsReBar/files/14233722/FARADIS-ROG.txt)

[NvStrapsReBar data.txt](https://github.com/terminatorul/NvStrapsReBar/files/14233723/NvStrapsReBar.data.txt)

[LocalPciGPU.h.txt](https://github.com/terminatorul/NvStrapsReBar/files/14233726/LocalPciGPU.h.txt)

**Comment by terminatorul** [comment]
Your modified LocalPciGPU.h header has the right values, but there is an extra backtick character at the end of line for `TARGET_BRIDGE_PCI_FUNCTION`. You should remove it.

To revert UEFI settings back to stock, you must first disable ReBAR settings in `NvStrapsReBar.exe`. Disable both the GPU BAR size and the PCI BAR size in the menu.

Or you can also set back the current year in UEFI setup to a value before 2024, and reboot. You can change the year back to the right value after reboot.

**Comment by Druboo666** [comment]
what happened to **_LocalPciGPU.h_** file?
Without that file how to configure it for the graphics card
also getting this error when trying building
`build.py...
c:\users\NAME\edk2\MdePkg\MdePkg.dec(32): error 000E: File/directory not found in workspace
        c:\users\NAME\edk2\MdePkg\Library\MipiSysTLib\mipisyst\library\include`

**Comment by terminatorul** [comment]
The latest version can detect the needed GPU settings automatically using `NvStrapsReBar.exe`. The header file LocalPciGpu.h is not used any more.

I'll have to look at the error later, can you please delete or move away the `edk2` directory, and clone everything again and rebuild ?

[Edit: I ran the buld from scratch, works as expected. Let me know if you still have problems with the build error]

**Comment by Druboo666** [comment]
this is what i am getting

```
GenFw" -e DXE_DRIVER -o c:\users\faradis\edk2\Build\NvStrapsReBar\RELEASE_VS2019\X64\NvStrapsReBar\ReBarDxe\ReBarDxe\OUTPUT\NvStrapsReBar.efi c:\users\faradis\edk2\Build\NvStrapsReBar\RELEASE_VS2019\X64\NvStrapsReBar\ReBarDxe\ReBarDxe\DEBUG\NvStrapsReBar.dll
'GenFw' is not recognized as an internal or external command,
operable program or batch file.
NMAKE : fatal error U1077: '"GenFw" -e DXE_DRIVER -o c:\users\faradis\edk2\Build\NvStrapsReBar\RELEASE_VS2019\X64\NvStrapsReBar\ReBarDxe\ReBarDxe\OUTPUT\NvStrapsReBar.efi c:\users\faradis\edk2\Build\NvStrapsReBar\RELEASE_VS2019\X64\NvStrapsReBar\ReBarDxe\ReBarDxe\DEBUG\NvStrapsReBar.dll ' : return code '0x1'
Stop.


build.py...
 : error 7000: Failed to execute command
        C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.38.33130\bin\Hostx86\x86\nmake.exe /nologo tbuild [c:\users\faradis\edk2\Build\NvStrapsReBar\RELEASE_VS2019\X64\NvStrapsReBar\ReBarDxe\ReBarDxe]


build.py...
 : error F002: Failed to build module
        c:\users\faradis\edk2\NvStrapsReBar\ReBarDxe\ReBarDxe.inf [X64, VS2019, RELEASE]

- Failed -
Build end time: 22:54:18, Feb.12 2024
Build total time: 00:01:10

Build failed 
```

**Reply by terminatorul** [reply]
You need to start the x86 Native Tools Command Prompt from your start menu, and run the build in that window.
Also in that window you need to run the command `edksetup.bat Rebuild` in the edk2 directory.

Follow the build instructions, and attach the full build output if you still have errors

**Comment by Druboo666** [comment]
how to insert bios modules with MMTOOL 4.5.0.23.As my motherboard is ASUS Z170 series they prone to break Overclocking functionality if modded with  UEFI-TOOL.
i cant find "PCI BUS" which is said in original guide but i am seeing this

https://github.com/terminatorul/NvStrapsReBar/assets/2793013/11119840-7d03-4668-97b6-b7b1e4b55319

**Reply by terminatorul** [reply]
Just find the module GUID ... it's also in the instructions

**Reply by terminatorul** [reply]
So ... open the UEFI image in UEFITool, find the pciBus module, read the GUID, find the GUID In MMOTool

**Reply by Druboo666** [reply]
i cant save the image also the insert file goes into the **free space** of bios and i think its breaking the cap file.
and it produces a 0 kb file
![image](https://github.com/terminatorul/NvStrapsReBar/assets/2793013/93c8a9e5-7c1b-4378-a4ae-149455cf0ea6)
the guide of vol 03 index 18A from MMTool is shown in uefi tool
![image](https://github.com/terminatorul/NvStrapsReBar/assets/2793013/d18d2968-250b-435b-a4c9-b0cf23ab7047)
UPDATE: It seems my MMTool can't save any images on my system even if it is stock
It just throws the error 
What should I do?

**Reply by terminatorul** [reply]
Sorry, I never used MMOTool. I don't know what the problem is

**Reply by Druboo666** [reply]
@xCuri0 can you help me please 🙏🙏🙏🙏
@terminatorul MMTOOL 5.0.2 beta seems to save the image but i  dont know if that is safe?also i dont know if that making pad file issue

**Reply by terminatorul** [reply]
Still don't know what to say about MMTool, I never had the pad file problem, despite some padding warnings in UEFITool...

But luckily your motherboard has USB BIOS Flashback, meaning you can recover the BIOS even if you flash a bad image and your board can not POST. So you can just flash your image anyway.

Be carefull as I had a motherboard that could not recover this way. MSI M-Flash never triggered. So you may want to test the feature in advance. But I believe my experience with that board was an MSI problem...

Do you have some real information about your worries that UEFITool can break overclocking ... it sounds from another world to me, it just cannot be true

**Reply by Druboo666** [reply]
https://github.com/xCuri0/ReBarUEFI/issues/98 
thats in common issues and fixes

**Reply by terminatorul** [reply]
Right, thanks... Did you decide to flash the MB ? Do you have ReBAR now ?
